src/Auth0.AspNetCore.Authentication/ServiceCollectionExtensions.cs

```diff
@@ -24,17 +24,40 @@ public static Auth0WebAppAuthenticationBuilder AddAuth0WebAppAuthentication(this
         /// Add Auth0 configuration using Open ID Connect
         /// </summary>
         /// <param name="services">The original <see href="https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.iservicecollection">IServiceCollection</see> instance</param>
+        /// <param name="authenticationScheme">The name that will be use for the authentication scheme</param>
         /// <param name="configureOptions">A delegate used to configure the <see cref="Auth0WebAppOptions"/></param>
         /// <returns>The <see cref="Auth0WebAppAuthenticationBuilder"/> instance that has been created.</returns>
         public static Auth0WebAppAuthenticationBuilder AddAuth0WebAppAuthentication(this IServiceCollection services, string authenticationScheme, Action<Auth0WebAppOptions> configureOptions)
         {
+            return services.AddAuth0WebAppAuthentication(authenticationScheme, true, configureOptions);
+        }
+
+        /// <summary>
+        /// Add Auth0 configuration using Open ID Connect
+        /// </summary>
+        /// <param name="services">The original <see href="https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.iservicecollection">IServiceCollection</see> instance</param>
+        /// <param name="authenticationScheme">The name that will be use for the authentication scheme</param>
+        /// <param name="setDefaultSchemes">If true, the default authentication, signin, and challenge schemes will be set to the cookie scheme</param>
+        /// <param name="configureOptions">A delegate used to configure the <see cref="Auth0WebAppOptions"/></param>
+        /// <returns>The <see cref="Auth0WebAppAuthenticationBuilder"/> instance that has been created.</returns>
+        public static Auth0WebAppAuthenticationBuilder AddAuth0WebAppAuthentication(this IServiceCollection services, string authenticationScheme, bool setDefaultSchemes, Action<Auth0WebAppOptions> configureOptions)
+        {
+            if (setDefaultSchemes)
+            {
+                services
+                    .AddAuthentication(options =>
+                    {
+                        var authOptions = new Auth0WebAppOptions();
+
+                        configureOptions(authOptions);
+
+                        options.DefaultAuthenticateScheme = authOptions.CookieName;
+                        options.DefaultSignInScheme = authOptions.CookieName;
+                        options.DefaultChallengeScheme = authOptions.CookieName;
+                    });
+            }
```

Comment:
I dont think we need this change. If you do not want to our SDK to call `AddAuthentication`, you should not use our ServiceCollectionExtensions, you should use our `AuthenticationBuilderExtensions`. 

Instead, we should focus on just the cookieName in this code. I believe we should be dropping the `setDefaultSchemes` and rollback most of these changes but only ensure we set the cookie to `auth0Options.CookieName`.